spec/rspec/support/differ_spec.rb

```diff
@@ -261,6 +261,48 @@ def inspect; "<BrokenObject>"; end
           expect(diff).to be_diffed_as(expected_diff)
         end
 
+      context 'Time objects inside hashes' do
+        let(:time) { Time.utc(1969, 12, 31, 19, 01, 40, 101) }
+        let(:formatted_time) { ObjectInspector.inspect(time) }
+
+        it "outputs unified diff message of two hashes with Time object keys" do
+          expected_diff = %Q{
+@@ -1,2 +1,2 @@
+-#{formatted_time} => "b",
++#{formatted_time} => "c",
+}
+
+          diff = differ.diff({ time => 'c'}, { time => 'b' })
+          expect(diff).to be_diffed_as(expected_diff)
+        end
+
+        it "outputs unified diff message of two hashes with hashes inside them" do
+          expected_diff = %Q{
+@@ -1,2 +1,2 @@
+-{"key" => #{formatted_time}, "key1" => "value"} => "b",
++{"key" => #{formatted_time}, "key1" => "value"} => "c",
+}
+
+          left_side_hash = {{'key' => time, 'key1' => 'value'} => 'c'}
+          right_side_hash = {{'key' => time, 'key1' => 'value'} => 'b'}
+          diff = differ.diff(left_side_hash, right_side_hash)
+          expect(diff).to be_diffed_as(expected_diff)
+        end
```

Comment:
It seems odd/nonstandard to use a hash as a key inside a hash.  It's far more common to have nested hashes used as values inside a hash.  Does this spec cover any code paths that using the nested hash as a value instead wouldn't?  If not, I think it would be nice to change this to be closer to what typical user usage will be.